### PR TITLE
Bundle most types into globals.d.ts

### PR DESCRIPTION
**Pull request by kitsonk**
Refs: #632 

This PR is a step towards generating a single definition file for all types in deno.

For the dynamically generated types files, they are now bundled into a single file and inlined in the `assets["globals.d.ts"]` object.  There are still 3 external type files that are not part of the bundle:

- `typescript.d.ts` - Likely should always be inlined directly.
- `text-encoding.d.ts` - This is the types for a node module import that we need the types for.  It maybe difficult to inline.
- `fetch_types.d.ts` - These types are written as ambient/global types.  Further changes in another PR to make this into a UMD/modular definition file could be done so that it can be renamed `fetch_types.ts` and then bundled like other modules.

### CODE REVIEW

**Comment by kitsonk** [comment]
For the first, they were leaking anyways and hacking around them was dangerous. I found some strange things that TypeScript was just swallowing, like TextEncode and TextDecode. We were referencing the instance type, not the constructor type in globals. My thinking is that we find ways not to leak them after this PR and then they will just disappear.

For the failures, there maybe lint issues, but the Travis issue is more complex. I need to update the `rootDirs` on the generated type build, but tsc doesn't accept it on the command line.

File: rollup.config.js [hunk 1]
```diff
@@ -18,9 +18,7 @@ const typescriptPath = `${
 // the current build path
 const tsconfigOverride = {
   compilerOptions: {
-    paths: {
-      "*": ["*", path.join(process.cwd(), "*")]
-    }
+    rootDirs: ["js", path.join(process.cwd(), "gen")]
```
Comment (ry) [inline rollup.config.js]: I see.. so we're not rooted in the gen dir? What is the purpose of this?

File: js/fetch.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2018 the Deno authors. All rights reserved. MIT license.
+// tslint:disable:no-reference
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply js/fetch.ts]: Nope... I was experimenting with something and forgot to remove it.  Thanks.

File: rollup.config.js (hunk 1 above)
Comment (kitsonk) [reply rollup.config.js]: This instructs TypeScript to resolve relative paths as equals from these roots... it is specifically designed to deal with situations where you generate some stuff in another path, but want it treated like it were part of the static path.

Saying this out loud again though to try to get the paths in the types to make better sense, but in the end, I still couldn't get it to work, and I had to use the `/// <amd-module name="name"/>` for the modules we "care" about having sensible names.  So I may actually back this out and go back to the `paths` solution which put it in `gen`.  That is going to be far easier than re-writing the `tsconfig.generated.json` on the fly.

**Comment by kitsonk** [comment]
@ry this should now be ready...  As I said before, we can figure out ways post this to not leak types that we don't want to leak, and I will get rid of `lib.globals.d.ts` after this, as well as hopefully bundle the `lib.*.d.ts` files.

File: js/os.ts [hunk 3]
```diff
@@ -128,8 +128,8 @@ export function readFileSync(filename: string): Uint8Array {
   return new Uint8Array(dataArray!);
 }
 
-function createEnv(_msg: fbs.EnvironRes): { [index:string]: string } {
-  const env: { [index:string]: string } = {};
+function createEnv(_msg: fbs.EnvironRes): { [index: string]: string } {
```
Comment (kitsonk) [inline js/os.ts]: hmmm... I have prettier deal with things on save, and didn't pay attention to this reformatting, but it is valid, though not related to this PR.

File: js/types.ts [hunk 4]
```diff
@@ -8,6 +8,7 @@ export interface ModuleInfo {
   outputCode: string | null;
 }
 
+// tslint:disable:max-line-length
```
Comment (kitsonk) [inline js/types.ts]: tslint is not being applied to `.d.ts` files at all, even ones we author internally...  So when I renamed this, tslint picked up some linting errors.

**Pull request merged**